test/e2e/upgradetest/framework/framework.go

```diff
@@ -138,6 +139,45 @@ func (f *Framework) UpgradeOperator() error {
 	return err
 }
 
+func (f *Framework) WaitOperatorElected() error {
+	var pod v1.Pod
+	err := retryutil.Retry(5*time.Second, 2, func() (bool, error) {
+		opt := metav1.ListOptions{LabelSelector: "name=etcd-operator"}
+		pods, err := f.KubeCli.CoreV1().Pods(f.KubeNS).List(opt)
+		if err != nil {
+			return false, err
+		}
+		if len(pods.Items) == 0 {
+			return false, nil
+		}
+		pod = pods.Items[0]
```

Comment:
Assuming it will only have one pod. We don't need to make this too generic since we only implement what's needed in testing.